### ISSUE TITLE
[BUG] crc setup broken on Windows with user name containg space when using experimental-features

### ISSUE THREAD

**Issue by jeffmaury** [code-ready/crc#1338]
### General information

  * OS: Windows
  * Hypervisor: Hyper-V
  * Did you run `crc setup` before starting it (Yes/No)? Yes
  * Running CRC on: Laptop / Baremetal-Server / VM: Laptop

## CRC version
```bash
# Put `crc version` output here
CodeReady Containers version: 1.12.0+6710aff
OpenShift version: 4.4.8 (embedded in binary)
```
  
## CRC status
```bash
# Put `crc status --log-level debug` output here
DEBU CodeReady Containers version: 1.12.0+6710aff
DEBU OpenShift version: 4.4.8 (embedded in binary)
Machine 'crc' does not exist. Use 'crc start' to create it.
```

## CRC config
```bash
# Put `crc config view` output here
```

## Host Operating System
```bash
# Put the output of `cat /etc/os-release` in case of Linux
# put the output of `sw_vers` in case of Mac
# Put the output of `systeminfo` in case of Windows
Nom de l’hôte:                              DESKTOP-U2EVGBM                                                                                           
Nom du système d’exploitation:              Microsoft Windows 10 Professionnel                                                                        
Version du système:                         10.0.18363 N/A version 18363                                                                              
Fabricant du système d’exploitation:        Microsoft Corporation                                                                                     
Configuration du système d’exploitation:    Station de travail autonome                                                                               
Type de version du système d’exploitation:  Multiprocessor Free                                                                                       
Propriétaire enregistré:                    Jeff MAURY                                                                                                
Organisation enregistrée:                                                                                                                             
Identificateur de produit:                  00330-80819-37372-AA630                                                                                   
Date d’installation originale:              18/11/2019, 15:37:22                                                                                      
Heure de démarrage du système:              11/06/2020, 09:01:18                                                                                      
Fabricant du système:                       LENOVO                                                                                                    
Modèle du système:                          20LAS3NJ0H                                                                                                
Type du système:                            x64-based PC                                                                                              
Processeur(s):                              1 processeur(s) installé(s).                                                                              
                                            [01] : Intel64 Family 6 Model 142 Stepping 10 GenuineIntel ~1910 MHz                                      
Version du BIOS:                            LENOVO N27ET36W (1.22 ), 04/07/2019                                                                       
Répertoire Windows:                         C:\WINDOWS                                                                                                
Répertoire système:                         C:\WINDOWS\system32                                                                                       
Périphérique d’amorçage:                    \Device\HarddiskVolume1                                                                                   
Option régionale du système:                fr;Français (France)                                                                                      
Paramètres régionaux d’entrée:              fr;Français (France)                                                                                      
Fuseau horaire:                             (UTC+01:00) Bruxelles, Copenhague, Madrid, Paris                                                          
Mémoire physique totale:                    32 379 Mo                                                                                                 
Mémoire physique disponible:                6 731 Mo                                                                                                  
Mémoire virtuelle : taille maximale:        59 243 Mo                                                                                                 
Mémoire virtuelle : disponible:             13 451 Mo                                                                                                 
Mémoire virtuelle : en cours d’utilisation: 45 792 Mo                                                                                                 
Emplacements des fichiers d’échange:        C:\pagefile.sys                                                                                           
Domaine:                                    WORKGROUP                                                                                                 
Serveur d’ouverture de session:             \\DESKTOP-U2EVGBM                                                                                         
Correctif(s):                               13 Corrections installées.                                                                                
                                            [01]: KB4552931                                                                                           
                                            [02]: KB4497165                                                                                           
                                            [03]: KB4516115                                                                                           
                                            [04]: KB4517245                                                                                           
                                            [05]: KB4524569                                                                                           
                                            [06]: KB4528759                                                                                           
                                            [07]: KB4537759                                                                                           
                                            [08]: KB4538674                                                                                           
                                            [09]: KB4541338                                                                                           
                                            [10]: KB4552152                                                                                           
                                            [11]: KB4560959                                                                                           
                                            [12]: KB4561600                                                                                           
                                            [13]: KB4560960                                                                                           
Carte(s) réseau:                            5 carte(s) réseau installée(s).                                                                           
                                            [01]: Intel(R) Ethernet Connection (4) I219-LM                                                            
                                                  Nom de la connexion : Ethernet                                                                      
                                                  État :                Support déconnecté                                                            
                                            [02]: TAP-Windows Adapter V9                                                                              
                                                  Nom de la connexion : Ethernet 2                                                                    
                                                  DHCP activé :         Oui                                                                           
                                                  Serveur DHCP :        10.36.127.254                                                                 
                                                  Adresse(s) IP                                                                                       
                                                  [01]: 10.36.113.103                                                                                 
                                                  [02]: fe80::d527:58a7:4965:c2ad                                                                     
                                            [03]: Hyper-V Virtual Ethernet Adapter                                                                    
                                                  Nom de la connexion : vEthernet (Default Switch)                                                    
                                                  DHCP activé :         Non                                                                           
                                                  Adresse(s) IP                                                                                       
                                                  [01]: 172.17.134.145                                                                                
                                                  [02]: fe80::1dbc:68b3:1e6d:b893                                                                     
                                            [04]: Intel(R) Dual Band Wireless-AC 8265                                                                 
                                                  Nom de la connexion : Wi-Fi                                                                         
                                                  DHCP activé :         Oui                                                                           
                                                  Serveur DHCP :        192.168.0.254                                                                 
                                                  Adresse(s) IP                                                                                       
                                                  [01]: 192.168.0.42                                                                                  
                                                  [02]: fe80::8548:85c0:593:b8e9                                                                      
                                                  [03]: 2a01:e35:2ffe:3020:a02b:4519:8489:a51b                                                        
                                                  [04]: 2a01:e35:2ffe:3020:8548:85c0:593:b8e9                                                         
                                            [05]: Bluetooth Device (Personal Area Network)                                                            
                                                  Nom de la connexion : Connexion réseau Bluetooth                                                    
                                                  État :                Support déconnecté                                                            
Configuration requise pour Hyper-V:         Un hyperviseur a été détecté. Les fonctionnalités nécessaires à Hyper-V ne seront pas affichées.          
```

### Steps to reproduce

  1. `crc setup --enable-experimental-features`
  2. 
  3. 
  4. 

### Expected

OK status

### Actual

```
INFO Checking if oc binary is cached                                                                                                                    
INFO Checking if podman remote binary is cached                                                                                                         
INFO Checking if goodhosts binary is cached                                                                                                             
INFO Checking if CRC bundle is cached in '$HOME/.crc'                                                                                                   
INFO Checking if running as normal user                                                                                                                 
INFO Checking Windows 10 release                                                                                                                        
INFO Checking if Hyper-V is installed and operational                                                                                                   
INFO Checking if user is a member of the Hyper-V Administrators group                                                                                   
INFO Checking if Hyper-V service is enabled                                                                                                             
INFO Checking if the Hyper-V virtual switch exist                                                                                                       
INFO Found Virtual Switch to use: Default Switch                                                                                                        
INFO Checking if user is allowed to log on as a service                                                                                                 
INFO Will run as admin: Running secedit export command                                                                                                  
INFO Enabling service log on for user                                                                                                                   
FATA Unable to get sid for username: Jeff MAURY: Get-LocalUser : Impossible de trouver un paramètre positionnel acceptant l'argument « MAURY ».         
Au caractère Ligne:1 : 2                                                                                                                                
+ (Get-LocalUser -Name Jeff MAURY).Sid.value                                                                                                            
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                       
    + CategoryInfo          : InvalidArgument : (:) [Get-LocalUser], ParameterBindingException                                                          
    + FullyQualifiedErrorId : PositionalParameterNotFound,Microsoft.PowerShell.Commands.GetLocalUserCommand                                             
                                                                                                                                                        
: exit status 1                                                                                                                                         
```

### Logs

Before gather the logs try following if that fix your issue
```bash
$ crc delete -f
$ crc cleanup
$ crc setup
$ crc start --log-level debug
```

Please consider posting the output of `crc start --log-level debug`  on http://gist.github.com/ and post the link in the issue.

**Comment by anjannath** [comment]
@jeffmaury Thanks for the issue, the username should've been enclosed in `""` to be properly parsed when it contains a space.

Can you please give this binary(https://722-174116223-gh.circle-artifacts.com/0/crc/windows-amd64/crc.exe) a try it uses os/user package's `user.Uid` to get the sid, and should fix this bug, Thanks!

**Comment by jeffmaury** [comment]
setup worked for me with this binary but I can't start the cluster, don't know how to provide the pull secret file

**Comment by anjannath** [comment]
>setup worked for me with this binary but I can't start the cluster, don't know how to provide the pull secret file
Another concern for me if that it asks many times to go though UAC which is not user friendly

We are working on reducing the number of UAC prompts #1327 and a way to configure the pull secret and bundle

there are few manual steps right now, for start to work form the tray, you need to configure the pull-secret and bundle
`crc config set pull-secret-file /path/to/pull-secret` and for the bundle `crc config set bundle ~/.crc/cache/crc_hyperv_xxx.crcbundle`

**Comment by anjannath** [comment]
Fixed in #1353
